editor/src/clj/editor/code/lang/json.clj

```diff
@@ -0,0 +1,43 @@
+;; Copyright 2020-2022 The Defold Foundation
+;; Copyright 2014-2020 King
+;; Copyright 2009-2014 Ragnar Svensson, Christian Murray
+;; Licensed under the Defold License version 1.0 (the "License"); you may not use
+;; this file except in compliance with the License.
+;; 
+;; You may obtain a copy of the License, together with FAQs at
+;; https://www.defold.com/license
+;; 
+;; Unless required by applicable law or agreed to in writing, software distributed
+;; under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+;; CONDITIONS OF ANY KIND, either express or implied. See the License for the
+;; specific language governing permissions and limitations under the License.
+
+(ns editor.code.lang.json
+  (:require [clojure.data.json :as json]
+            [editor.code.data :as data])
+  (:import [java.io PushbackReader]))
+
+(set! *warn-on-reflection* true)
+(set! *unchecked-math* :warn-on-boxed)
+
+(def grammar
+  {:name "JSON"
+   :scope-name "source.json"
+   ;; indent patterns shamelessly stolen from textmate:
+   ;; https://github.com/textmate/json.tmbundle/blob/master/Preferences/Miscellaneous.tmPreferences
+   :indent {:begin #"^.*(\{[^}]*|\[[^\]]*)$"
+            :end #"^\s*[}\]],?\s*$"}
+   :patterns [{:match #"\b(?:true|false|null)\b"
+               :name "constant.language.json"}
+              {:match #"(?x)-?(?:0|[1-9]\d*)(?:\n(?:\n\.\d+)?(?:[eE][+-]?\d+)?)?"
```

Comment:
Did you write it yourself? What does the `(?x)` mean at the front? I tried inputting this regex in an online regex visualizer, and it was shown as invalid with that `(?x)` prefix...

Reply:
I believe we lifted it from the TextMate patterns? I just moved it here from the `editor.json` module I deleted in this PR.

Anyway, https://regex101.com reports it as a modifier flag for the remainder of the expression. Looks like it enables comments or something? It should be supported in Java 8 and above according to the site.
```
(?x) match the remainder of the pattern with the following effective flags: gmx
     x modifier: extended. Spaces and text after a # in the pattern are ignored
```